proto/proto-backplane-grpc/src/main/proto/deephaven/proto/console.proto

```diff
@@ -131,21 +143,36 @@ message AutoCompleteRequest {
         // Requests that a response be sent back with completion items
         GetCompletionItemsRequest get_completion_items = 3;
 
+        // Request for help about the method signature at the cursor
+        GetSignatureHelpRequest get_signature_help = 7;
+
+        // Request for help about what the user is hovering over
+        GetHoverRequest get_hover = 8;
+
+        // Request to perform file diagnostics
+        GetDiagnosticRequest get_diagnostic = 9;
+
         // Closes the document, indicating that it will not be referenced again
         CloseDocumentRequest close_document = 4;
     }
 }
 message AutoCompleteResponse {
+    int32 request_id = 2;
+    bool success = 3;
     oneof response {
         GetCompletionItemsResponse completion_items = 1;
+        GetSignatureHelpResponse signatures = 4;
+        GetHoverResponse hover = 5;
+        GetPullDiagnosticResponse diagnostic = 6;
+        GetPublishDiagnosticResponse diagnostic_publish = 7;
     }
 }
 
 message BrowserNextResponse {
 }
 
 message OpenDocumentRequest {
-    io.deephaven.proto.backplane.grpc.Ticket console_id = 1;
+    io.deephaven.proto.backplane.grpc.Ticket console_id = 1 [deprecated=true];
```

Comment:
Should we have a server safety layer that ensures _if_ the deprecated fields and new fields are set, they are equal; otherwise blowup? Alternatively, we could say either the new fields must be set, or the deprecated fields must be set, but not both.

Reply:
I think the only reason for deprecation here (as opposed to just ripping out these fields) is if a user had an older web client but newer engine version. Realistically I think the only place that may happen right now is with web developers, so I would be fine just removing these fields instead of deprecating, but I know @niloc132 had other thoughts on that